lib/src/chart/radar_chart/radar_chart_painter.dart

```diff
@@ -0,0 +1,310 @@
+        );
+      });
+
+      path.close();
+      canvas.drawPath(path, _graphPaint);
+      canvas.drawPath(path, _graphBorderPaint);
+    });
+  }
+
+  @override
+  RadarTouchResponse handleTouch(FlTouchInput touchInput, Size size) {
+    if (size == null || (dataSetsPosition?.isEmpty ?? true)) return null;
+
+    final touchedSpot = _getNearestTouchSpot(size, touchInput.getOffset(), dataSetsPosition);
+    return RadarTouchResponse(touchedSpot, touchInput);
+  }
+
+  double radarCenterY(Size size) => size.height / 2.0;
+
+  double radarCenterX(Size size) => size.width / 2.0;
+
+  double radarRadius(Size size) => min(radarCenterX(size), radarCenterY(size)) * 0.8;
+
+  RadarTouchedSpot _getNearestTouchSpot(
+    Size viewSize,
+    Offset touchedPoint,
+    List<RadarDataSetsPosition> radarDataSetsPosition,
+  ) {
+    for (int i = 0; i < radarDataSetsPosition.length; i++) {
+      final dataSetPosition = radarDataSetsPosition[i];
+      for (int j = 0; j < dataSetPosition.entriesOffset.length; j++) {
+        final entryOffset = dataSetPosition.entriesOffset[j];
+        if ((touchedPoint.dx - entryOffset.dx).abs() <=
+                targetData.radarTouchData.touchSpotThreshold &&
+            (touchedPoint.dy - entryOffset.dy).abs() <=
+                targetData.radarTouchData.touchSpotThreshold) {
+          return RadarTouchedSpot(
+            targetData.dataSets[i],
+            i,
+            targetData.dataSets[i].dataEntries[j],
+            j,
+            FlSpot(entryOffset.dx, entryOffset.dy),
+            entryOffset,
+          );
+        }
+      }
+    }
+    return null;
+  }
+
+  List<RadarDataSetsPosition> _calculateDataSetsPosition(Size viewSize) {
+    if (viewSize == null || data?.dataSets == null) return null;
+
+    final centerX = radarCenterX(viewSize);
+    final centerY = radarCenterY(viewSize);
+    final radius = radarRadius(viewSize);
+
+    final scale = radius / data.maxEntry.value;
+    final angle = (2 * pi) / data.titleCount;
+
+    final dataSetsPosition = List<RadarDataSetsPosition>(data.dataSets.length);
+    for (int i = 0; i < data.dataSets.length; i++) {
+      final dataSet = data.dataSets[i];
+      final entriesOffset = List<Offset>(dataSet.dataEntries.length);
+
+      for (int j = 0; j < dataSet.dataEntries.length; j++) {
+        final point = dataSet.dataEntries[j];
+
+        final xAngle = cos(angle * j - pi / 2);
+        final yAngle = sin(angle * j - pi / 2);
+        final scaledPoint = scale * point.value;
+
+        final entryOffset = Offset(
+          centerX + scaledPoint * xAngle,
+          centerY + scaledPoint * yAngle,
+        );
+
+        entriesOffset[j] = entryOffset;
+      }
+      dataSetsPosition[i] = RadarDataSetsPosition(entriesOffset);
+    }
+
+    return dataSetsPosition;
+  }
+
+  @override
+  bool shouldRepaint(RadarChartPainter oldDelegate) {
+    final repaint = oldDelegate.data != data;
+    log('repaint radar chart: $repaint');
```

Comment:
If you think it is working correctly please remove it :))

Reply:
:)